CMakeLists.txt

```diff
@@ -77,67 +80,26 @@ if(MSVC AND CAF_SANITIZERS)
   message(FATAL_ERROR "Sanitizer builds are currently not supported on MSVC")
 endif()
 
-# -- compiler setup ------------------------------------------------------------
+# -- base target setup ---------------------------------------------------------
+
+# This target propagates compiler flags, extra dependencies, etc. All other CAF
+# targets pull this target in as PRIVATE dependency. Users that embed CAF into
+# their own CMake scaffold (e.g., via FetchContent) may pass this target in with
+# some properties predefined in order to force compiler flags or dependencies.
```

Comment:
If this is a recommendation we should document it somewhere outside of this comment.

Reply:
Good point. I think this should go into a cohesive CMake/build guide, though. The wiki might be a good place for that.

Reply:
This does not stand in the way of merging this PR right?

Reply:
Don't think so. 😊